lib/galaxy/jobs/__init__.py

```diff
@@ -366,7 +366,7 @@ def __init__(self, app: MinimalManagerApp):
             else:
                 job_config_file = self.app.config.job_config_file
                 if not self.app.config.is_set("job_config_file") and not os.path.exists(job_config_file):
-                    job_config_file = os.path.join(os.path.dirname(self.app.config.config_file), "job_conf.xml")
+                    job_config_file = os.path.join(self.app.config.config_dir, "job_conf.xml")
                     if os.path.exists(job_config_file):
```

Comment:
```suggestion
                    old_default_job_config_file = os.path.join(self.app.config.config_dir, "job_conf.xml")
                    if os.path.exists(old_default_job_config_file):
                        job_config_file = old_default_job_config_file
```

I think it would be better to set `job_config_file` to the old default file *only if* that file exists, so that if neither of the default files is found, the `log.warning()` in the `except OSError:` branch complains with `Job configuration "job_conf.yml" does not exist, using default job configuration`.